### PR TITLE
Plasma update to 6.1.1

### PR DESCRIPTION
**Pull request by Luciogi**
Wait for https://github.com/void-linux/void-packages/pull/51007

#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-GLIBC)
# TODO
- [x] krdp (make new package)
- [x] drkonqi

### CODE REVIEW

**Comment by Luciogi** [comment]
- [ ] kdeplasma-addons cross building fails
<details><summary>Details</summary>
```
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /builddir/.xbps-kdeplasma-addons/wrappers/armv7l-linux-gnueabihf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /builddir/.xbps-kdeplasma-addons/wrappers/armv7l-linux-gnueabihf-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Installing in the same prefix as Qt, adopting their path scheme.
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Failed
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Found OpenGL: /usr/armv7l-linux-gnueabihf/usr/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /usr/armv7l-linux-gnueabihf/usr/lib32/libxkbcommon.so (found suitable version "1.7.0", minimum required is "0.5.0")
-- Found WrapVulkanHeaders: /usr/armv7l-linux-gnueabihf/usr/include
-- Found KF6Config: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Config/KF6ConfigConfig.cmake (found version "6.3.0")
-- Found KF6CoreAddons: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6CoreAddons/KF6CoreAddonsConfig.cmake (found version "6.3.0")
-- Found KF6DBusAddons: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6DBusAddons/KF6DBusAddonsConfig.cmake (found version "6.3.0")
-- Found KF6Declarative: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Declarative/KF6DeclarativeConfig.cmake (found version "6.3.0")
-- Found KF6GlobalAccel: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6GlobalAccel/KF6GlobalAccelConfig.cmake (found version "6.3.0")
-- Found KF6Holidays: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Holidays/KF6HolidaysConfig.cmake (found version "6.3.0")
-- Found Gettext: /usr/bin/msgmerge (found version "0.22.5")
-- Found KF6I18n: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6I18n/KF6I18nConfig.cmake (found version "6.3.0")
CMake Error at /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Auth/KF6AuthConfig.cmake:62 (include):
  include could not find requested file:

    KAUTH_TARGETSFILE-NOTFOUND
Call Stack (most recent call first):
  /usr/share/ECM/find-modules/FindKF6.cmake:52 (find_package)
  CMakeLists.txt:51 (find_package)


-- Found KF6Auth: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Auth/KF6AuthConfig.cmake (found version "6.3.0")
-- Found X11: /usr/armv7l-linux-gnueabihf/usr/include
-- Looking for XOpenDisplay in /usr/armv7l-linux-gnueabihf/usr/lib/libX11.so;/usr/armv7l-linux-gnueabihf/usr/lib/libXext.so
-- Looking for XOpenDisplay in /usr/armv7l-linux-gnueabihf/usr/lib/libX11.so;/usr/armv7l-linux-gnueabihf/usr/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Found KF6KIO: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6KIO/KF6KIOConfig.cmake (found version "6.3.0")
-- Found KF6KCMUtils: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6KCMUtils/KF6KCMUtilsConfig.cmake (found version "6.3.0")
-- Found KF6Notifications: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Notifications/KF6NotificationsConfig.cmake (found version "6.3.0")
-- Found KF6Runner: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Runner/KF6RunnerConfig.cmake (found version "6.3.0")
-- Found KF6Service: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Service/KF6ServiceConfig.cmake (found version "6.3.0")
-- Found KF6Sonnet: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6Sonnet/KF6SonnetConfig.cmake (found version "6.3.0")
-- Found KF6UnitConversion: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6UnitConversion/KF6UnitConversionConfig.cmake (found version "6.3.0")
-- Found KF6XmlGui: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6XmlGui/KF6XmlGuiConfig.cmake (found version "6.3.0")
-- Found KF6NewStuff: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6NewStuff/KF6NewStuffConfig.cmake (found version "6.3.0")
-- Found KF6JobWidgets: /usr/armv7l-linux-gnueabihf/usr/lib32/cmake/KF6JobWidgets/KF6JobWidgetsConfig.cmake (found version "6.3.0")
-- Found KF6: success (found suitable version "6.3.0", minimum required is "6.2.0") found components: Config CoreAddons DBusAddons Declarative GlobalAccel Holidays I18n Auth KIO KCMUtils Notifications Runner Service Sonnet UnitConversion XmlGui NewStuff JobWidgets
-- Could NOT find KirigamiAddons (missing: KirigamiAddons_DIR)
-- Could NOT find KItemModels (missing: KItemModels_DIR)
-- Found the following ICU libraries:
--   uc (required): /usr/armv7l-linux-gnueabihf/usr/lib/libicuuc.so
--   i18n (required): /usr/armv7l-linux-gnueabihf/usr/lib/libicui18n.so
-- Found ICU: /usr/armv7l-linux-gnueabihf/usr/include (found suitable version "73.2", minimum required is "66.1")
-- Could NOT find Qt6Quick3D (missing: Qt6Quick3D_DIR)
-- Could NOT find Qt6WebEngineQuick (missing: Qt6WebEngineQuick_DIR)
-- Could NOT find Qt6WebEngineQuick (missing: Qt6WebEngineQuick_DIR)
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- The following OPTIONAL packages have been found:

 * OpenGL
 * Vulkan
 * Qt6Test (required version >= 6.6.0)
 * Qt6Core5Compat (required version >= 6.6.0)
 * Qt6WidgetsTools (required version >= 6.6.0)
 * Qt6Widgets (required version >= 6.5.0)
 * Qt6CoreTools (required version >= 6.6.0)
 * Qt6DBusTools (required version >= 6.6.0)
 * Qt6QmlTools (required version >= 6.6.0)
 * XKB (required version >= 0.5.0), XKB API common to servers and clients., <http://xkbcommon.org>
 * WrapVulkanHeaders
 * Qt6GuiTools (required version >= 6.6.0)
 * Freetype
 * PkgConfig
 * Fontconfig
 * ICU (required version >= 66.1), Unicode and Globalization support for software applications
   Provides alternate calendar systems that are not available in QCalendar

-- The following REQUIRED packages have been found:

 * KF6DBusAddons (required version >= 6.2.0)
 * KF6Declarative (required version >= 6.2.0)
 * KF6GlobalAccel (required version >= 6.2.0)
 * KF6Holidays (required version >= 6.2.0)
 * Gettext
 * KF6I18n (required version >= 6.2.0)
 * KF6Auth (required version >= 6.2.0)
 * KF6KIO (required version >= 6.2.0)
 * KF6KCMUtils (required version >= 6.2.0)
 * KF6Notifications (required version >= 6.2.0)
 * KF6Runner (required version >= 6.2.0)
 * KF6Sonnet (required version >= 6.2.0)
 * KF6UnitConversion (required version >= 6.2.0)
 * KF6XmlGui (required version >= 6.2.0)
 * KF6NewStuff (required version >= 6.2.0)
 * KF6JobWidgets (required version >= 6.2.0)
 * KF6 (required version >= 6.2.0)
 * Qt6Core (required version >= 6.5.0)
 * ECM (required version >= 5.240.0)
 * Qt6HostInfo
 * Qt6
 * Plasma5Support (required version >= 6.1.0)
 * Qt6Qml
 * Qt6Gui (required version >= 6.5.0)
 * Qt6Quick
 * Plasma
 * PlasmaQuick (required version >= 6.1.0)

-- The following RUNTIME packages have not been found:

 * KF6Purpose, Framework for cross-application services and actions, <https://commits.kde.org/purpose>
   Needed for QuickShare applet
 * KirigamiAddons, Extra controls for Kirigami applications
   Required at runtime for the User List widget
 * KItemModels, Set of item models extending the Qt model-view framework
   Required at runtime for the Dictionary, Kate Profiles, and Konsole Profiles applets
 * Qt6Quick3D, A high-level API for creating 3D content and 3D user interfaces based on Qt Quick
   Required for desktop cube

-- The following OPTIONAL packages have not been found:

 * Qt6QmlCompilerPlusPrivate
 * Qt6WebEngineQuick (required version >= 6.6.0), <https://doc.qt.io/qt-6/qtwebengine-index.html>
   Needed by the dict applet
   Needed by the web browser applet
 * Git

-- Configuring incomplete, errors occurred!
=> ERROR: kdeplasma-addons-6.1.0_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${build_wrksrc}' exited with 1
=> ERROR:   in do_configure() at common/build-style/cmake.sh:77
```
</details>

**Comment by Luciogi** [comment]
# Big issues
- [ ] In Wayland session, plasmashell is crashing in every 5 seconds

File: srcpkgs/kdeplasma-addons/template [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
 maintainer="Luciogi <lucigithubcommit@skiff.com>"
 license="GPL-2.0-or-later, LGPL-2.1-or-later"
 homepage="https://invent.kde.org/plasma/kdeplasma-addons"
 distfiles="${KDE_SITE}/plasma/${version}/${pkgname}-${version}.tar.xz"
-checksum=911af4754873c17c1ca7d49f0669de3e386858b09adf282783b9c5510576ba95
+checksum=cb57fa427b3188a1bc7a6b12e196fc11252742e08f6397e3af00ffaf34676127
+
+if [ $CROSS_BUILD ];then
+	hostmakedepends+=" kf6-kauth-devel"
+fi
```
Comment (sgn) [inline srcpkgs/kdeplasma-addons/template:25]: No, put `kf6-kauth-tools` in `hostmakedepends` instead.

Comment (sgn) [reply srcpkgs/kdeplasma-addons/template:25]: Also add `qt6-webenegine-devel` on where applicable.

File: srcpkgs/freerdp/template [hunk 2]
```diff
@@ -36,6 +36,10 @@ post_install() {
 subpackages="libfreerdp-server freerdp-server freerdp-server-devel libfreerdp
  freerdp-devel"
 
+post_install() {
+	sed -i -e 's;[$]{_IMPORT_PREFIX}/bin;/usr/bin;' ${DESTDIR}/usr/lib/cmake/WinPR2/WinPRTargets-*.cmake
+}
+
```
Comment (sgn) [inline srcpkgs/freerdp/template:42]: Drop this commit, u did it wrong. See c9080fbf7ac9f1a4778a24917f135b4a18c29ec2

**Review by sgn — CHANGES_REQUESTED** [review]

**Comment by sgn** [comment]
Also: https://invent.kde.org/network/kaccounts-integration/-/merge_requests/61 https://invent.kde.org/network/kaccounts-providers/-/merge_requests/40

**Comment by sgn** [comment]
https://bugs.kde.org/show_bug.cgi?id=488794

Gonna remove `plasma-wayland-protocols` from development dependdencies, rebuild them to see if any pkg need it, add it into `makedepends` explicitly, rebuild everything.
Also KF5 package.

**Comment by Luciogi** [comment]
> https://bugs.kde.org/show_bug.cgi?id=488794
> 
> Gonna remove `plasma-wayland-protocols` from development dependdencies, rebuild them to see if any pkg need it, add it into `makedepends` explicitly, rebuild everything. Also KF5 package.

crashing issue was fixed

File: srcpkgs/kdeplasma-addons/template (hunk 1 above)
Comment (Luciogi) [reply srcpkgs/kdeplasma-addons/template:25]: I dont remember which pkgs need as optional dep qt6-webengine-devel

This can be done later

**Comment by Luciogi** [comment]
[ ] plasma 6.1.1 releases (needs minimium plasma-wayland-protocols 1.13)

**Comment by Luciogi** [comment]
updated to plasma 6.1.1

**Comment by Luciogi** [comment]
## Tested in VM:
- Wayland session is stable
- Screen recording working

**Comment by AlexPewMaster** [comment]
> shaking cursor when gets big, it become invisible, and as it restore to normal shape its visible again (nvidia)

I'm seeing the same issue on my side. Just updated to Plasma 6.1.1 and the cursor turns invisible whenever I shake my mouse. I have an AMD Radeon RX 580 graphics card.

### About this system
Operating System: Void 
KDE Plasma Version: 6.1.1
KDE Frameworks Version: 6.3.0
Qt Version: 6.7.2
Kernel Version: 6.6.35_1 (64-bit)
Graphics Platform: Wayland
Processors: 12 × AMD Ryzen 5 2600 Six-Core Processor
Memory: 31.3 GiB of RAM
Graphics Processor: AMD Radeon RX 580 Series

**Comment by Luciogi** [comment]
> > shaking cursor when gets big, it become invisible, and as it restore to normal shape its visible again (nvidia)
> 
> I'm seeing the same issue on my side. Just updated to Plasma 6.1.1 and the cursor turns invisible whenever I shake my mouse. I have an AMD Radeon RX 580 graphics card.
> ### About this system
> 
> Operating System: Void KDE Plasma Version: 6.1.1 KDE Frameworks Version: 6.3.0 Qt Version: 6.7.2 Kernel Version: 6.6.35_1 (64-bit) Graphics Platform: Wayland Processors: 12 × AMD Ryzen 5 2600 Six-Core Processor Memory: 31.3 GiB of RAM Graphics Processor: AMD Radeon RX 580 Series

this known bug, it will be fixed in 6.1.2

**Pull request closed without merging**